Library/Homebrew/livecheck/options.rb

```diff
@@ -0,0 +1,75 @@
+# typed: strong
+# frozen_string_literal: true
+
+module Homebrew
+  module Livecheck
+    # Options to modify livecheck's behavior. These primarily come from
+    # `livecheck` blocks but they can also be set by livecheck at runtime.
+    class Options < T::Struct
+      # Whether to use brewed curl.
+      const :homebrew_curl, T.nilable(T::Boolean)
+
+      # Form data to use when making a `POST` request.
+      const :post_form, T.nilable(T::Hash[T.any(String, Symbol), String])
+
+      # JSON data to use when making a `POST` request.
+      const :post_json, T.nilable(T::Hash[T.any(String, Symbol), String])
+
+      # Returns a `Hash` of options that are provided as arguments to `url`.
+      sig { returns(T::Hash[Symbol, T.untyped]) }
+      def url_options
+        {
+          homebrew_curl:,
+          post_form:,
+          post_json:,
+        }
+      end
+
+      # Returns a `Hash` of all instance variables, using `String` keys.
+      sig { returns(T::Hash[String, T.untyped]) }
+      def to_hash
+        T.let(serialize, T::Hash[String, T.untyped])
+      end
+
+      # Returns a `Hash` of all instance variables, using `Symbol` keys.
+      sig { returns(T::Hash[Symbol, T.untyped]) }
+      def to_h = to_hash.transform_keys(&:to_sym)
+
+      # Returns a new object formed by merging `other` values with a copy of
+      # `self`.
+      #
+      # `nil` values are removed from `other` before merging if it is an
+      # `Options` object, as these are unitiailized values. This ensures that
+      # existing values in `self` aren't unexpectedly overwritten with defaults.
+      sig { params(other: T.any(Options, T::Hash[Symbol, T.untyped])).returns(Options) }
+      def merge(other)
+        return dup if other.blank?
```

Comment:
```suggestion
        return dup if other.empty?
```
(marginally faster/clearer since it's directly defined on both param types)